Dockerfile

```diff
@@ -25,6 +25,13 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
 # add our user and group first to make sure their IDs get assigned consistently
 RUN groupadd -r freight && useradd -r -m -g freight freight
 
+# This is the build user on cheffed and salted host machines
+# that would run the Freight container. Directories that we volume
+# mount when running Freight are owned by 9010:9010 (build).
+# The Freight container is run with --user=9010:9010, but it needs to exist
+# in here for that to work.
+RUN groupadd -g 9010 build && useradd -r -g 9010 -u 9010 build
```

Comment:
what's the error message you get? usually you don't need the user explicitly in the container to run as it

```console
$ docker run --rm -ti --user $RANDOM ubuntu:focal bash -c 'echo $UID'
12524
```

Reply:
So.... this is not very good, but basically what happened was that I added `--user=9010:9010` and that results in a crashloop. Unfortunately on garbage, one does not simply view docker logs........

```
 docker logs dockerctl_freight_worker_00
Error response from daemon: configured logging driver does not support reading
```

So this is more of a shot in the dark.

Reply:
Okay wait, logs are written to syslog, you just can't read them with `docker logs` but they're there.

Reply:
This seems to be the issue, lemme look more into it.

`Aug  5 17:57:38 garbage 35961a728241[1387]: error: failed switching to "freight": operation not permitted`

Reply:
that sounds like `gosu` -- which we can probably just factor out?

Reply:
Ah, this is from `gosu freight`.

Reply:
I think I'd have to see what inside Freight's container relies on it running as freight... likely not much since previously it would basically have fabric login to garbage as `build` user.

Reply:
Opted to not factor out gosu for now... some files like `/srv/docker/freight/etc/freight/google-credentials.json` are root:root on Freight and we'd want to keep it that way. I inlined the auth helper script as a refactor and so you can see what's going on there - entrypoint runs as root to do some things like that and then steps down to `build` now before execing freight stuff.